File: src/main/java/io/openfuture/openmessanger/configuration/AwsS3Config.java
package io.openfuture.openmessanger.configuration;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import com.amazonaws.auth.AWSCredentials;
import com.amazonaws.auth.AWSStaticCredentialsProvider;
import com.amazonaws.auth.BasicAWSCredentials;
import com.amazonaws.regions.Regions;
import com.amazonaws.services.s3.AmazonS3;
import com.amazonaws.services.s3.AmazonS3ClientBuilder;

@Configuration
public class AwsS3Config {

    @Bean
    public AmazonS3 amazonS3(AWSCredentials awsCredentials) {
        return AmazonS3ClientBuilder
                .standard()
                .withCredentials(new AWSStaticCredentialsProvider(awsCredentials))
                .withRegion(Regions.US_WEST_2)
                .build();
    }

    @Bean
    public AWSCredentials awsCredentials() {
        return new BasicAWSCredentials(
                "AKIAUSWCDIFRDTGV52OL",
                "/2poY+Og6+p5ax1n8YR9WNVZD6KnrdbDDwrXq0FW"
        );
    }

}

File: src/main/kotlin/io/openfuture/openmessenger/configuration/AwsS3Config.kt
package io.openfuture.openmessenger.configuration

import com.amazonaws.auth.AWSCredentials
import com.amazonaws.auth.AWSStaticCredentialsProvider
import com.amazonaws.auth.BasicAWSCredentials
import com.amazonaws.regions.Regions
import com.amazonaws.services.s3.AmazonS3
import com.amazonaws.services.s3.AmazonS3ClientBuilder
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class AwsS3Config(
    private val awsConfig: AwsConfig
) {

    @Bean
    fun amazonS3(awsCredentials: AWSCredentials?): AmazonS3 {
        return AmazonS3ClientBuilder
            .standard()
            .withCredentials(AWSStaticCredentialsProvider(awsCredentials))
            .withRegion(Regions.US_EAST_2)
            .build()
    }

    @Bean
    fun awsCredentials(): AWSCredentials {
        return BasicAWSCredentials(
            awsConfig.accessKey,
            awsConfig.secretKey
        )
    }
}